bokehjs/src/coffee/models/plots/plot_canvas.coffee

```diff
@@ -631,12 +631,13 @@ class PlotCanvasView extends Renderer.View
   _map_hook: (ctx, frame_box) ->
 
   _paint_empty: (ctx, frame_box) ->
-    @visuals.border_fill.set_value(ctx)
-    ctx.fillRect(0, 0,  @canvas_view.mget('width'), @canvas_view.mget('height'))
-    ctx.clearRect(frame_box...)
-    @visuals.background_fill.set_value(ctx)
-    ctx.fillRect(frame_box...)
-
+    ctx.clearRect(0, 0,  @canvas_view.mget('width'), @canvas_view.mget('height'))
+    if @visuals.border_fill.doit
+      @visuals.border_fill.set_value(ctx)
+      ctx.fillRect(0, 0,  @canvas_view.mget('width'), @canvas_view.mget('height'))
+    if @visuals.background_fill.doit
```

Comment:
I don't know why the border area is called a border when in other places we use "border" to refer to outlines (similar to like css).